### PR TITLE
Multislice propagation

### PR DESCRIPTION
**Pull request by mdw771**
- Added simple multislice simulation module
- Placeholder assumed for probe.wavelength

### CODE REVIEW

File: xdesign/propagation.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    pass
+
+
+## Add energy / wavelength attribute to probe class?
+def _slice_modify(delta_slice, beta_slice, probe, wavefront, delta_nm, lmda):
+    """Modify wavefront within a slice.
+
+    Parameters:
+    -----------
+    delta_grid : ndarray
```
Comment (dgursoy) [inline xdesign/propagation.py]: delta_grid or delta_slice?

File: xdesign/propagation.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+    return wavefront
+
+
+def _slice_propagate(wavefront, delta_nm, lat_nm, shape, lmda):
+    """Free space propagation.
+
+    Parameters:
+    -----------
+
```
Comment (dgursoy) [inline xdesign/propagation.py]: missing.

**Comment by dgursoy** [comment]
Is this WIP?

**Comment by mdw771** [comment]
Some missing function annotations are fixed. Please see latest commit

**Pull request closed without merging**